tests/posix/common/src/pthread.c

```diff
@@ -678,24 +682,54 @@ ZTEST(posix_apis, test_sched_policy)
 	static const bool policy_enabled[] = {
 		IS_ENABLED(CONFIG_COOP_ENABLED),
 		IS_ENABLED(CONFIG_PREEMPT_ENABLED),
-		IS_ENABLED(CONFIG_PREEMPT_ENABLED),
+		IS_ENABLED(CONFIG_PREEMPT_ENABLED) && IS_ENABLED(CONFIG_COOP_ENABLED),
 		false,
 	};
-	static int nprio[] = {
-		CONFIG_NUM_COOP_PRIORITIES,
-		CONFIG_NUM_PREEMPT_PRIORITIES,
```

Comment:
Is it possible to still keep `nprio[]`? From what I can tell, only the `42` value would need to change.

`CONFIG_NUM_COOP_PRIORITIES + CONFIG_NUM_PREEMPT_PRIORITIES` looks good to me, which is (I think) where you are going.

Reply:
I just noticed that `nprio[]` is also not `const`, but it probably should be. Feel free to add that drive-by fix, if you're so inclined.

Reply:
Ok - I think i remember why this wasn't const before. Please ignore.